tasks/scripts.js

```diff
@@ -8,7 +8,7 @@ module.exports = (gulp, $, pkg) => {
       default: { concat: true },
     }), args);
     return gulp.src(pkg.gulpPaths.scripts.src)
-      .pipe($.plumber())
+      .pipe($.if(process.env.GULP_ABORT_ON_ERROR, $.eslint.failAfterError()))
```

Comment:
Currently we're using `minimist` to retrieve CLI options while hardly rely on environment variables. Does it makes sense to adjust the other options to use env variables as well by introducing [`dotenv`](https://www.npmjs.com/package/dotenv), at least for the `options.production` condition? 🤔

Reply:
Based on the comment by @sun we could add an option like this and check if it's set. If not, the pipe will break otherwise.

```
$ gulp build --continue

const options = Object.assign($.minimist(process.argv.slice(2), {
  string: ['outputStyle'],
  boolean: ['sourcemaps', 'production', 'continue'],
  default: { },
}), args);

.pipe($.if(options.continue, $.plumber()))
```

A Gulp task will fail if an error occurs. [Here's](https://gist.github.com/floatdrop/8269868) a small article about the behaviour.

We also need to introduce it for the scripts task as this is not handled by this PR yet.

Reply:
This sounds like the proper solution to me. Can we make `--continue` default to `true` via `package.json`?

Or change the suggested change here to explicitly check for `options.continue === false`?

Reply:
Just pushed two new commits, trying to integrate the proposed changes. I totally agree with @fabianmarz  that having flags AND an env value is odd, so I introduced an error handling flag `fail-after-error`. 
Also, having understood now what plumber does thanks to the article, I put it back where it belongs.

**What about `--continue`?**
As @sun pointed out, that's a flag to control gulp behavior. If there is a sequence of tasks (like in our `build` task, by default one of them failing will prevent the others from being run, unless `--continue` is provided. 
The way I see it, using the same flag name to control the error behavior of the linters (like `gulp scripts --continue=false`) would be highly confusing, as you could end up with something like `gulp --continue build --continue=false`. That's why I proposed to name our flag `--fail-after-error`. 

**Summing up:**
- The proposed change is backwards compatible. No flag – behavior does not change.
- `--fail-after-error` flag will make scripts and styles task exit with error code > 0 if they detect an error.
- documented in readme.md

Looking forward to getting your feedback!

Reply:
Happy to move forward with this for now. We can still revise it in a future version if necessary.  @fabianmarz What do you think?

Only the docs could use some tweaks, but I will try to add a fixup for those directly now.

Reply:
Agree, can be merged :)